Source/Core/VideoCommon/VertexLoaderX64.cpp

```diff
@@ -0,0 +1,458 @@
+#include "Common/CPUDetect.h"
+#include "Common/x64ABI.h"
+#include "VideoCommon/VertexLoaderX64.h"
+
+using namespace Gen;
+
+static const X64Reg src_reg = ABI_PARAM1;
+static const X64Reg dst_reg = ABI_PARAM2;
+static const X64Reg count_reg = ABI_PARAM3;
+static const X64Reg scratch1 = RAX;
+static const X64Reg scratch2 = R8;
+static const X64Reg scratch3 = R9;
+static const X64Reg skipped_reg = R10;
+
+VertexLoaderX64::VertexLoaderX64(const TVtxDesc& vtx_desc, const VAT& vtx_att): VertexLoaderBase(vtx_desc, vtx_att)
+{
+	if (!IsInitialized())
+		return;
+
+	AllocCodeSpace(4096);
+	ClearCodeSpace();
+	GenerateVertexLoader();
+	WriteProtect();
+}
+
+OpArg VertexLoaderX64::GetVertexAddr(int array, u64 attribute)
+{
+	OpArg data = MDisp(src_reg, m_src_ofs);
+	if (attribute & MASK_INDEXED)
+	{
+		if (attribute == INDEX8)
+		{
+			MOVZX(64, 8, scratch1, data);
+		}
+		else
+		{
+			MOVZX(64, 16, scratch1, data);
+			// Convert to little-endian.
+			ROR(16, R(scratch1), Imm8(8));
```

Comment:
We support BSWAP(16, R(X)) as an alias for rotate; might make the intent a bit more clear.  Also, minor performance tweak, you probably want to rotate then movzx to avoid the register merging penalty on Intel.